engine/consensus/matching/engine_test.go

```diff
@@ -57,6 +58,8 @@ func (s *MatchingEngineSuite) SetupTest() {
 	s.engine, err = NewEngine(unittest.Logger(), net, me, metrics, metrics, s.state, s.receipts, s.index, s.core)
 	require.NoError(s.T(), err)
 
+	ctx := irrecoverable.NewMockSignalerContext(s.T(), context.Background())
```

Comment:
Could you also add teardown logic to make sure we are gracefully stopping the engine?
- Use `NewMockSignalerContextWithCancel` and store the `CancelFunc` in the test suite struct
- Add a `TearDownTest` function, call `cancel()` and wait for `engine.Done` using `AssertClosesBefore`